### PR TITLE
Add Yarr egg

### PR DESCRIPTION
**Pull request by zoomiti**
# Description
This pull request adds a Yarr egg. [Yarr](https://github.com/nkanaev/yarr) is a web based rss feed aggregator. 

This egg compiles yarr from source so that it can support arm processors. This is mainly due to the fact that Yarr only has releases for amd64. I tried installing go using alpine in the install script but that did not work.
The best solution was to download go from their website, install it, and then use it to compile yarr.

## Checklist for all submissions

* [X] Have you followed the guidelines in our [Contributing document](https://github.com/parkervcp/eggs/blob/master/CONTRIBUTING.md)?
* [X] Have you checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change?
* [X] Have you tested and reviewed your changes with confidence that everything works?
* [X] Did you branch your changes and PR from that branch and not from your master branch?

## New egg Submissions

1. [X] Does your submission pass tests (server is connectable)?
2. [ ] Does your egg use a custom docker image?
    * [ ] Have you tried to use a generic image?
    * [ ] Did you PR the necessary changes to make it work?
3. [X] Have you added the egg to the main README.md and any other README files in subdirectories of the egg (e.g /game_eggs) according to the alphabetical order?
4. [X] Have you added a unique README.md for the egg you are adding according to the alphabetical order?

### CODE REVIEW

File: software/yarr/egg-yarr.json [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    "config": {
+        "files": "{}",
+        "startup": "{\r\n    \"done\": \"starting server\",\r\n    \"userInteraction\": []\r\n}",
+        "logs": "{}",
+        "stop": "^C"
+    },
+    "scripts": {
+        "installation": {
+            "script": "apt update\r\napt install -y git make gcc curl tar\r\n\r\ncd\r\n\r\nARCH=$([[ \"$(uname -m)\" == \"x86_64\" ]] && echo \"amd64\" || echo \"arm64\")\r\n\r\necho ${ARCH}\r\n\r\ncurl -O https:\/\/dl.google.com\/go\/go1.19.2.linux-${ARCH}.tar.gz\r\nls\r\ntar xvf go1.19.2.linux-${ARCH}.tar.gz\r\n\r\nchown -R root:root .\/go\r\nmv go \/usr\/local\r\n\r\nexport GOPATH=$HOME\/work\r\nexport PATH=$PATH:\/usr\/local\/go\/bin:$GOPATH\/bin\r\n\r\ncd \/mnt\/server\r\n\r\nmkdir yarr_source\r\ngit clone https:\/\/github.com\/nkanaev\/yarr.git yarr_source\r\n\r\ncd yarr_source\r\n\r\nexport CGO_ENABLED=1\r\nsed -i -r \"s\/amd64\/$ARCH\/g\" makefile\r\n\r\nmake build_linux    # -> _output\/linux\/yarr\r\n\r\ncp _output\/linux\/yarr ..\/yarr\r\ncd ..\/\r\n\r\nrm -rf yarr_source\r\n\r\ntouch auth.conf\r\n\r\nchmod +x yarr\r\n\r\necho \"Install complete\"",
```
Comment (QuintenQVD0) [inline software/yarr/egg-yarr.json:25]: Please just use a premade container image that has Golang installed and is Debian based `golang:1.19.2-bullseye`

File: software/yarr/egg-yarr.json [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+            "user_viewable": false,
+            "user_editable": true,
+            "rules": "required|string|max:20|alpha_dash",
+            "field_type": "text"
+        },
+        {
+            "name": "Password",
+            "description": "The password needed to log in to yarr",
+            "env_variable": "PASSWORD",
+            "default_value": "password",
```
Comment (QuintenQVD0) [inline software/yarr/egg-yarr.json:45]: Providing a default password is a no-go for security

**Review by QuintenQVD0 — CHANGES_REQUESTED** [review]

**Comment by zoomiti** [comment]
So as stated in their github page, yarr is feature complete and unlikely to commit anything that would break building from source. Does this mean this egg is fine as is? Or should I add a variable to checkout and build from a specific release where "latest" defaults to just compiling from the master branch?

File: software/yarr/egg-yarr.json [hunk 3]
```diff
@@ -22,8 +22,8 @@
...
-            "container": "debian:buster-slim",
+			"script": "cd \/mnt\/server\r\n\r\nmkdir yarr_source\r\ngit clone https:\/\/github.com\/nkanaev\/yarr yarr_source\r\n\r\ncd yarr_source\r\n\r\nmake build_default  # -> _output\/yarr\r\n\r\ncp _output\/yarr ..\/yarr\r\ncd ..\/\r\n\r\nrm -rf yarr_source\r\n\r\nchmod +x yarr\r\n\r\necho \"Install complete\"",
```
Comment (QuintenQVD0) [inline software/yarr/egg-yarr.json:25]: where is the `#!/bin/bash` at the start of the install script?

**Review by QuintenQVD0 — CHANGES_REQUESTED** [review]

File: software/yarr/egg-yarr.json [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    "description": "Yarr is a selfhosted feed aggregator",
+    "features": null,
+    "docker_images": {
+        "ghcr.io\/parkervcp\/yolks:debian": "ghcr.io\/parkervcp\/yolks:debian"
+    },
+    "file_denylist": [],
+    "startup": "echo \"{{USERNAME}}:{{PASSWORD}} > .\/auth.conf && .\/yarr -addr 0.0.0.0:{{SERVER_PORT}} -auth-file=.\/auth.conf -db=.\/feed.sql",
+    "config": {
+        "files": "{}",
+        "startup": "{\r\n    \"done\": \"starting server\",\r\n    \"userInteraction\": []\r\n}",
```
Comment (QuintenQVD0) [inline software/yarr/egg-yarr.json:19]: userInteraction is not more used

**Review by QuintenQVD0 — CHANGES_REQUESTED** [review]

File: software/yarr/egg-yarr.json (hunk 4 above)
Comment (zoomiti) [reply software/yarr/egg-yarr.json:19]: is it safe to just remove that line or whats your actual suggestion here? i'm not sure how to resolve this

Comment (QuintenQVD0) [reply software/yarr/egg-yarr.json:19]: Just remove that line. And remove the, at the end of the line before

**Review by QuintenQVD0 — APPROVED** [review]

**Pull request merged**